file.go

```diff
@@ -33,7 +37,16 @@ func downloadGithubZipFile(gitHubCommit GitHubCommit, gitHubToken string, instan
 		return zipFilePath, wrapError(err)
 	}
 
-	resp, err := httpClient.Do(req)
+	for retries +=1; retries > 0; retries -= 1 {
+		resp, err := httpClient.Do(req)
+		if err != nil {
```

Comment:
I can't recall if the Go HTTP library returns an error for non 200 responses? Or if it returns a response and you need to check the status code? Probably worth verifying.

Also, we should only retry on _some_ types of errors. E.g., Retrying on 3xx and 4xx is probably not worth it, but retries on 5xx, timeouts, and connection issues probably is.

Reply:
Potentially huge discussion here, which I've had before a few times and there isn't a hugely correct single answer. The  nuance here is that 3xx isn't an _error_ as such, and 4xx is deemed to be a _client_ error, while 5xx is a _server_ error.

In fact, I've had a few situations in the past where a 404 error has actually been successfully retried (due to the underlying storage eventually converging over the lifetime of the retry cycle) and also where 3xx should definitely not have been retried (since it was at that stage an infinite loop).

However, the semantics here are probably "good enough", in that things which aren't definitely successes will be retried, and indeed there's some (quite small) chance the error will be resolved at the end of the chain.

References: https://en.wikipedia.org/wiki/List_of_HTTP_status_codes

Reply:
Yup, agreed it's just a heuristic and not perfect!